skip_iterator.java

```diff
@@ -0,0 +1,66 @@
+// Time Complexity : O(1)
+// Space Complexity : O(1)
+// Did this code successfully run on Leetcode : Yes
+// Any problem you faced while coding this : No
+
+
+// Your code here along with comments explaining your approach
+public class Main {
+    public static class SkipIterator implements Iterator<Integer>{
+        Iterator<Integer> it;
+        HashMap<Integer,Integer> map;
+        Integer nextEl;
+
+        public SkipIterator(Iterator<Integer> it){
+            this.it = it;
+            map = new HashMap<>();
+            advance();
+        }
+        public Integer next(){
+            Integer result = nextEl;
+            advance();
+            return result;
+        }
+        public boolean hasNext(){
```

Comment:
Only for hasNEXT TC is constant other 2 methods is having time complexity = O(N)